airbyte-integrations/connectors/source-slack/source_slack/components.py

```diff
@@ -0,0 +1,114 @@
+from dataclasses import dataclass
+from typing import Optional
+from typing import List, Mapping, Any, Iterable
+import requests
+from airbyte_cdk.models import AirbyteMessage, SyncMode, Type
+from airbyte_cdk.sources.declarative.types import Config, Record, StreamSlice, StreamState
+from airbyte_cdk.sources.declarative.extractors import DpathExtractor
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation, AddFields
+from airbyte_cdk.sources.declarative.partition_routers import SubstreamPartitionRouter
+
+
+@dataclass
+class ChannelMembersExtractor(DpathExtractor):
+    """
+    Transform response from list of strings to list dicts:
+    from: ['aa', 'bb']
+    to: [{'member_id': 'aa'}, {{'member_id': 'bb'}]
+    """
+    def extract_records(self, response: requests.Response) -> List[Record]:
+        records = super().extract_records(response)
+        return [{'member_id': record} for record in records]
+
+
+@dataclass
+class JoinChannels(RecordTransformation):
+    """
+    Make 'conversations.join' POST request for every found channel id
+    if we are not still a member of such channel
+    """
+
+    def transform(
+        self,
+        record: Record,
+        config: Optional[Config] = None,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+    ) -> Record:
+        # The `is_member` property indicates whether or not the API Bot is already assigned / joined to the channel.
+        # https://api.slack.com/types/conversation#booleans
+        channel_id = record.get('id')
+        if config.get('join_channels') and not record.get("is_member"):
+            response = requests.post(
+                url='https://slack.com/api/conversations.join',
+                headers={
+                    'Content-Type': 'application/json',
+                    'Authorization': f'Bearer {config["api_token"]}'
+                },
+                params={'channel': channel_id}
+            )
+            print(response.json())
+
+            # WHAT TO DO IF IT FAILS ????????????????????????
```

Comment:
Skip with the log message "The channel: ____ could not be joined, <possible_issues_goes_here>, skipping."